src/sinks/aws_sqs.rs

```diff
@@ -0,0 +1,433 @@
+        let fifo = config.queue_url.ends_with(".fifo");
+
+        let sqs = SqsSink {
+            client,
+            queue_url: config.queue_url,
+        };
+
+        let sink = request
+            .batch_sink(
+                SqsRetryLogic,
+                sqs,
+                VecBuffer::new(batch.size),
+                batch.timeout,
+                cx.acker(),
+            )
+            .sink_map_err(|e| error!("Fatal sqs sink error: {}", e))
+            .with_flat_map(move |e| iter_ok(encode_event(e, &encoding, fifo)));
+
+        Ok(sink)
+    }
+}
+
+impl Service<Vec<SendMessageBatchRequestEntry>> for SqsSink {
+    type Response = SendMessageBatchResult;
+    type Error = RusotoError<SendMessageBatchError>;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, entries: Vec<SendMessageBatchRequestEntry>) -> Self::Future {
+        debug!(
+            message = "sending records.",
+            events = %entries.len(),
+        );
+
+        let sizes: Vec<usize> = entries
+            .iter()
+            .map(|entry| entry.message_body.len())
+            .collect();
+
+        let client = self.client.clone();
+        let request = SendMessageBatchRequest {
+            entries,
+            queue_url: self.queue_url.clone(),
+        };
+
+        Box::pin(async move {
+            client
+                .send_message_batch(request)
+                .inspect_ok(|_| {
+                    for byte_size in sizes {
+                        emit!(AwsSqsEventSent { byte_size });
+                    }
+                })
+                .instrument(info_span!("request"))
+                .await
+        })
+    }
+}
+
+impl EncodedLength for SendMessageBatchRequestEntry {
+    fn encoded_length(&self) -> usize {
+        // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessageBatch.html
+        // Request example:
+        // https://sqs.us-east-2.amazonaws.com/123456789012/MyQueue/
+        // ?Action=SendMessageBatch
+        // &SendMessageBatchRequestEntry.1.Id=test_msg_001
+        // &SendMessageBatchRequestEntry.1.MessageBody=test%20message%20body%201
+        // &SendMessageBatchRequestEntry.2.Id=test_msg_002
+        // &SendMessageBatchRequestEntry.2.MessageBody=test%20message%20body%202
+        // &SendMessageBatchRequestEntry.2.DelaySeconds=60
+        // &SendMessageBatchRequestEntry.2.MessageAttribute.1.Name=test_attribute_name_1
+        // &SendMessageBatchRequestEntry.2.MessageAttribute.1.Value.StringValue=test_attribute_value_1
+        // &SendMessageBatchRequestEntry.2.MessageAttribute.1.Value.DataType=String
+        // &Expires=2020-05-05T22%3A52%3A43PST
+        // &Version=2012-11-05
+        // &AUTHPARAMS
+
+        let length_prefix = 32; // SendMessageBatchRequestEntry.10.
+
+        length_prefix
```

Comment:
Why is the encoded length not just `self.message_body.len()`? I would expect the batch size and AWS's API limits to be specific to the payload sizes rather than including metadata.

Reply:
I did not found in the documentation what exactly included in the payload limit, so decide to add the length of fields. But I think it's okay to change to `self.message_body.len()` because I set the limit to 250kB instead 256KB, so we should be good.

Reply:
:+1: 

> The maximum allowed individual message size and the maximum total payload size (the sum of the individual lengths of all of the batched messages) are both 256 KB (262,144 bytes). 

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessageBatch.html

Makes me think that it is just the messages themselves that count towards the limit.